### DISCUSSION TITLE
Override the generator

### DISCUSSION THREAD

**Discussion by shainegordon** [orval-labs/orval#620] [Q&A] [answered]
So I am currently using a custom implementation of OpenAPI spec -> react query which is based on NSwag.CodeGeneration. 
it works well, but has a lot of custom code. 

I’d like to migrate to Orval, but there is currently a blocker. 

ALL of our endpoints are POST’s. We basically auto generate an “RPC-style” API of business operations. 
A developer can then build a server side operation, and it becomes available in the front end, fully typed - it’s a fantastic DX. 

Because every endpoints ends up as a POST, but on the server a developer explicitly implements a Query or a Command, we are able to add a “tag” to the open api spec - “queryType=query/command”
Our code gen then is able to determine is it should be creating a query or a mutation based on this tag, and it works great. 

I assume I can just import GENERATOR_CLIENT into my own code, and then add my own property like “react-query-custom” where I can then just spread the existing “react-query” and override the “client” function used?

**Comment by anymaniax** [comment, marked as answer]
Hello @shainegordon, now you have a @orval/core so you can do whatever you want. You can also use [this](https://orval.dev/reference/configuration/output#query) and if it doesn't feed your need we can still discuss

**Reply by iamguid** [reply]
Can you please check this discussion https://github.com/anymaniax/orval/discussions/843
